stdlib/public/core/BridgeObjectiveC.swift

```diff
@@ -376,9 +376,22 @@ public struct AutoreleasingUnsafeMutablePointer<Pointee /* TODO : class */>
   public var pointee: Pointee {
     /// Retrieve the value the pointer points to.
     @_transparent get {
-      // We can do a strong load normally.
-      return UnsafePointer(self).pointee
+      // The memory addressed by this pointer contains a non-owning reference,
+      // therefore we *must not* point an `UnsafePointer<AnyObject>` to
+      // it---otherwise we would allow the compiler to assume it has a +1
+      // refcount, enabling some optimizations that wouldn't be valid.
```

Comment:
This makes sense but is kinda terrifying. I need to think carefully about some of the NSDictionary bridging stuff…

Reply:
All +0 references in memory are a hazard; they must only be handled through the `Unmanaged` glovebox.  E.g., we'll need to audit any and all NSFastEnumeration implementations & usages from Swift code.

(Currently Unmanaged doesn't quite get this right either, but Michael is already working on fixing that.)